docs/setup_installation/azure/getting_started.md

```diff
@@ -24,78 +24,137 @@ To run all the commands on this page the user needs to have at least the followi
+az identity create --name $UA_IDENTITY_NAME --resource-group $RESOURCE_GROUP
+
+export UA_IDENTITY_PRINCIPAL_ID=`az identity show --name $UA_IDENTITY_NAME --resource-group $RESOURCE_GROUP --query principalId --output tsv`
+export UA_IDENTITY_CLIENT_ID=`az identity show --name $UA_IDENTITY_NAME --resource-group $RESOURCE_GROUP --query clientId --output tsv`
+export UA_IDENTITY_RESOURCE_ID=`az identity show --name $UA_IDENTITY_NAME --resource-group $RESOURCE_GROUP --query id --output tsv`
 ```
 
-### Step 1.4: Retrieve setup Identifiers
+### Step 1.4: Grant permissions to the User-Assigned Managed Identity
 
-Create a set of environment variables for use in later steps.
+Create a custom role definition with the minimum permissions needed to read and write to the storage account:
 
 ```bash
-export managed_id=`az aks show --resource-group $resource_group --name $cluster_name --query "identity.principalId" --output tsv`
+export STORAGE_ID=`az storage account show --name $STORAGE_ACCOUNT_NAME --resource-group $RESOURCE_GROUP --query "id" --output tsv`
+
+az role definition create --role-definition '{
+  "Name": "hopsfs-storage-permissions",
+  "IsCustom": true,
+  "Description": "Allow HopsFS to access the storage container",
+  "Actions": [
+    "Microsoft.Storage/storageAccounts/blobServices/containers/write",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/read",
+    "Microsoft.Storage/storageAccounts/blobServices/write",
+    "Microsoft.Storage/storageAccounts/blobServices/read",
+    "Microsoft.Storage/storageAccounts/listKeys/action"
+  ],
+  "NotActions": [],
+  "DataActions": [
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/delete",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/read",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/move/action",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/write"
+  ],
+  "AssignableScopes": [
+    "'$STORAGE_ID'"
+  ]
+}'
+
+sleep 30  # give Azure some time to persist the new role
+
+az role assignment create --role hopsfs-storage-permissions --assignee $UA_IDENTITY_PRINCIPAL_ID --scope $STORAGE_ID
+```
 
-export storage_id=`az storage account show --name $storage_account_name --resource-group $resource_group --query "id" --output tsv`
+Also, assign roles to pull/push images from/to the container registry:
 
-export acr_id=`az acr show --name $registry_name --resource-group $resource_group --query "id" --output tsv`
+```bash
+az role assignment create --assignee $UA_IDENTITY_PRINCIPAL_ID --role AcrPull --scope $ACR_ID
+az role assignment create --assignee $UA_IDENTITY_PRINCIPAL_ID --role "AcrPush" --scope $ACR_ID
+az role assignment create --assignee $UA_IDENTITY_PRINCIPAL_ID --role "AcrDelete" --scope $ACR_ID
```

Comment:
do you still need to do that? given that we use service principle to connect to the container registry 
and for the aks cluster itself we attach the acr registry so it can pull from there?

Reply:
I've just tested it and it is indeed unnecessary. Fixed.